lib/inlines.js

```diff
@@ -977,9 +981,9 @@ var EMOTICONS = [
 ];
 var parseEmoticon = function(block) {
     var m;
-    if ((m = this.tryMatch(reEmoticon))) {
+    if ((m = this.tryMatch(reEmoticon))  && !reDelimHttp.test(this.subject)) {
```

Comment:
I don't think `reDelimHttp` is actually needed, so we can remove it. We look for links before we check for emoticons, and we only call `parseEmoticon` when the string starts with a colon or less than symbol, so `reDelimHttp` should never find a match.

Also, that pattern only matches `http://` and we allow users to add custom URL protocols, so I don't think this would've worked any way. It was a good idea to check this though.

Reply:
hi @hmhealey the check is to prevent message such as `http://example.com/:)/` to render emoji, i added the test too https://github.com/mattermost/commonmark.js/pull/9/commits/e3d3a489126f9ffcfd031eaf54a2b5716ca9d92b.

Regarding the custom url protocol, will it render as link too?

Reply:
It could be rendered as a link, but that is controlled by the `urlFilter` that gets passed in and used [here](https://github.com/mattermost/commonmark.js/blob/master/lib/inlines.js#L858).

I just tested this code without `reDelimHttp`, and it passes the test that you added in that commit, so it's safe to remove it.

Reply:
hi @hmhealey okay i have remove the `reDelimHttp`, thanks